lib/queue.js

```diff
@@ -398,10 +398,17 @@ Queue.prototype._setupQueueEventListeners = function() {
     const key = channel.split('@')[0];
     switch (key) {
       case progressKey: {
-        const commaPos = message.indexOf(',');
-        const jobId = message.substring(0, commaPos);
-        const progress = message.substring(commaPos + 1);
-        this.emit('global:progress', jobId, JSON.parse(progress));
+        // New way to send progress message data
+        try {
+          const { progress, jobId } = JSON.parse(message);
+          this.emit('global:progress', jobId, JSON.parse(progress));
```

Comment:
Do we need this second `JSON.parse`, since we are sending `progress` this time, not `progressJSON` string.

Reply:
oh yes, you are right, some test should have caught this error though, I will fix it.